### PR TITLE
Add sizing restrictions to generated go code

### PR DESCRIPTION
**Pull request by nullstyle**

### CODE REVIEW

**Comment by nullstyle** [comment]
Some notes about this implementation:

Struct fields use the `xdrmaxsize` tag to specify their length.  Additionally, typedefs can contain sized declarations and we support this by making the typedef confirm to the `Sized` interface which go-xdr uses to decide max size.

**Comment by raymens** [comment]
@nullstyle What is the status of this issue?

**Comment by nullstyle** [comment]
@raymens It's stalled.  None of us have looked at it since I originally implemented it.  I imagine I'll take another look at integrating it when I get around to reworking the xdr package or doing something else related to xdrgen.

File: lib/xdrgen/generators/go.rb [hunk 1]
```diff
@@ -501,6 +523,25 @@ def switch_for(out, union, ident)
...
+        case field.declaration
+        when Xdrgen::AST::Declarations::String
+          size = field.declaration.resolved_size
+        when Xdrgen::AST::Declarations::Array
+          size = field.declaration.resolved_size unless field.declaration.fixed?
+        end
+
+        if size == "MAX_TX_PER_LEDGER"
+          binding.pry
+        end
```
Comment (bartekn) [inline lib/xdrgen/generators/go.rb]: This `if` can now be removed.

**Pull request merged**